dll/win32/msgina/shutdown.c

```diff
@@ -637,72 +754,76 @@ CreateToolTipForButtons(
 }
 
 VOID
-ReplaceRequiredButton(
+EndFriendlyDialog(
     HWND hDlg,
-    HINSTANCE hInstance,
-    BOOL bIsAltKeyPressed,
-    BOOL bIsSleepButtonReplaced)
+    PSHUTDOWN_DLG_CONTEXT pContext)
 {
-    int destID = IDC_BUTTON_SLEEP;
-    int targetedID = IDC_BUTTON_HIBERNATE;
-    HWND hwndDest, hwndTarget;
-    RECT rect;
-    WCHAR szBuffer[30];
-
-    /* If the sleep button has been already replaced earlier, bring sleep button back to its original position */
-    if (bIsSleepButtonReplaced)
+    MSGINACLOSETHEMEDATA msCloseThemeData;
+    int i;
+    
+    if (pContext->hUxtheme)
     {
-        destID = IDC_BUTTON_HIBERNATE;
-        targetedID = IDC_BUTTON_SLEEP;
+        msCloseThemeData = (MSGINACLOSETHEMEDATA)GetProcAddress(pContext->hUxtheme, "CloseThemeData");
+        if (!msCloseThemeData)
+            return;
```

Comment:
In this case you are not killing the timer and removing the subclass, is that intentional?

Reply:
Nope.

Reply:
but still not fixed?

Reply:
Better late than never